pygments/lexers/mcfunction.py

```diff
@@ -0,0 +1,272 @@
+"""
+    pygments.lexers.mcfunction
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexers for MCFunction and related languages.
+
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, default, include, using
+from pygments.token import (Comment, Error, Keyword, Literal, Name, Number,
+                            Operator, Punctuation, String, Text, Token,
+                            Whitespace)
+
+
+__all__ = ['SNBTLexer', 'MCFunctionLexer']
+
+
+class SNBTLexer(RegexLexer):
+    """Lexer for stringified NBT, a data format used in Minecraft
+
+
+    .. versionadded:: ?
+    """
+
+    name = "SNBT"
+    url = ""
+    aliases = ["snbt"]
+    filenames = ["*.snbt"]
+    mimetype = ["text/snbt"]
+
+    tokens = {
+        "root": [
+            # We only look for the open bracket here since square bracket
+            #  is only valid in NBT pathing (which is a mcfunction idea).
+            (r"(?=\{)", Punctuation, "compound"),
+        ],
+
+        "whitespace": [
+            (r"\s+", Whitespace),
+        ],
+
+        "operators": [
+            (r"[,:;]", Punctuation),
+        ],
+
+        "literals": [
+            (r"(true|false)", Keyword.Constant),
+            (r"(-?\d*\.\d+[fFdD]?)", Number.Float),
+            (r"(-?\d+[bBsSlLfFdD]?)", Number.Integer),
+            (r"(-?\d+[eE]-?\d+)", Number.Float),
+            # Seperate states for both types of strings so they don't entangle
+            (r'"', String.Double, "literals.string_double"),
+            (r"'", String.Single, "literals.string_single"),
+        ],
+        "literals.string_double": [
+            (r"\\.", String.Escape),
+            (r'[^\\"\n]+', String.Double),
+            (r'"', String.Double, "#pop"),
+            (r"[$\n]", Error, "#pop"),
```

Comment:
So literal strings can't contain a dollar character? Don't you rather mean that they must be matched and not run till the end of the line (`$`)? A `$` character in a character class is escaped, it has the literal meaning of a dollar.